kani-compiler/src/kani_middle/transform/kani_intrinsics.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+//! Module responsible for generating code for a few Kani intrinsics.
+//!
+//! These intrinsics have code that depend on information from the compiler, such as type layout
+//! information; thus, they are implemented as a transformation pass where their body get generated
+//! by the transformation.
+
+use crate::kani_middle::attributes::matches_diagnostic;
+use crate::kani_middle::transform::body::{CheckType, MutableBody, SourceInstruction};
+use crate::kani_middle::transform::check_values::{build_limits, ty_validity_per_offset};
+use crate::kani_middle::transform::{TransformPass, TransformationType};
+use crate::kani_queries::QueryDb;
+use rustc_middle::ty::TyCtxt;
+use stable_mir::mir::mono::Instance;
+use stable_mir::mir::{
+    BinOp, Body, Constant, Operand, Place, Rvalue, Statement, StatementKind, RETURN_LOCAL,
+};
+use stable_mir::target::MachineInfo;
+use stable_mir::ty::{Const, RigidTy, TyKind};
+use std::fmt::Debug;
+use strum_macros::AsRefStr;
+use tracing::trace;
+
+/// Generate the body for a few Kani intrinsics.
```

Comment:
Is this struct designed only for valid value check? If yes, we might we want renamed the struct name to reflect it. Otherwise the comment here can be used to list all check body generated by this pass.

Reply:
It is designed to be generic, but valid value check is the only use case we have right now.

Reply:
This is fairly internal to the compiler, so let me know if you have a name you prefer. Otherwise, this is something that we can revisit later.